src/components/v5/common/Pills/types.ts

```diff
@@ -0,0 +1,41 @@
+import { MessageDescriptor } from 'react-intl';
+import { SimpleMessageValues } from '~types';
+
+export type ExtensionStatusBadgeMode =
+  | 'coming-soon'
+  | 'not-installed'
+  | 'enabled'
+  | 'disabled'
+  | 'deprecated'
+  | 'governance'
+  | 'new'
+  | 'staking'
+  | 'finalizable'
+  | 'claimed'
+  | 'payments';
+
+export type PillMode =
+  | 'dedicated'
+  | 'dedicatedFilled'
+  | 'active'
+  | 'activeFilled'
+  | 'new'
+  | 'activeNew'
+  | 'top'
+  | 'topFilled'
+  | 'banned'
+  | 'team';
+
+export type IconSize = 'extraTiny' | 'tiny';
+
+export type PillSize = 'medium' | 'small';
+
+export type PillsProps = {
```

Comment:
Maybe would be better to have separate interfaces for PillsBase and ExtensionStatusBadge?

```
export interface PillsProps {  
  mode?: PillMode;
  text?: MessageDescriptor | string;
  textValues?: SimpleMessageValues;
  iconName?: string;
  iconSize?: IconSize;
  pillSize?: PillSize;
  className?: string;
}
```

and 

```
interface ExtensionStatusBadgeProps extends Omit<PillsProps, mode> {
 mode?: ExtensionStatudBadgeMode
}
```